frontend/javascripts/oxalis/view/right-border-tabs/tree_hierarchy_view.tsx

```diff
@@ -238,17 +239,24 @@ function TreeHierarchyView(props: Props) {
     }
   }
 
-  function setExpansionOfAllSubgroupsTo(group: TreeNode, expanded: boolean) {
-    const selectedGroupIdKey = getNodeKeyFromNode(group);
-    const subGroupKeys = deepFlatFilter([group], (node) => node.type === GroupTypeEnum.GROUP).map(
-      (node) => node.key,
-    );
-
-    if (expanded) {
-      setExpandedNodeKeys(_.uniq([...expandedNodeKeys, ...subGroupKeys]));
-    } else {
-      setExpandedNodeKeys(_.without(expandedNodeKeys, selectedGroupIdKey, ...subGroupKeys));
+  function setExpansionOfAllSubgroupsTo(parentGroup: TreeNode, expanded: boolean) {
+    if (parentGroup.id === MISSING_GROUP_ID) {
+      const newGroups = mapGroups(props.treeGroups, (group) => {
+        return { ...group, isExpanded: expanded };
```

Comment:
same here (the code will be longer but it's always good to avoid object creations)